installers/macos-pkg-setup-template.sh

```diff
@@ -69,8 +69,9 @@ fi
 chmod +x ../python $PYTHON_MAJOR $PYTHON_MAJOR_DOT_MINOR $PYTHON_MAJOR_MINOR python
 
 echo "Upgrading pip..."
+export PIP_ROOT_USER_ACTION=ignore
 ./python -m ensurepip
-./python -m pip install --ignore-installed pip --disable-pip-version-check --no-warn-script-location --root-user-action=ignore
+./python -m pip install --ignore-installed pip --disable-pip-version-check --no-warn-script-location
```

Comment:
Note that this same issue occurs for linux runners: https://github.com/jmarrec/python-versions/actions/runs/8250002087/job/22564591071#step:6:30

Reply:
On Linux where python is built from sources, the issue occurs for non latest patch versions or EOL versions.
All latest patch versions for python 3.7+ are embedding pip>=22.1 which support this options.